File: clients/dart/cldf/lib/clid/models/validation_result.dart
/// Validation result for ID generation
class ValidationResult {
  final List<String> errors;
  final List<String> warnings;

  const ValidationResult({this.errors = const [], this.warnings = const []});

  bool get isValid => errors.isEmpty;

  @override
  String toString() {
    if (isValid) {
      return warnings.isEmpty
          ? 'Valid'
          : 'Valid with warnings: ${warnings.join(', ')}';
    }
    return 'Invalid: ${errors.join(', ')}';
  }
}

File: clients/java/cldf-java/src/main/java/app/crushlog/cldf/globalid/ValidationResult.java
package app.crushlog.cldf.globalid;

import java.util.ArrayList;
import java.util.List;

import lombok.Getter;

/** Validation result for ID generation */
@Getter
public class ValidationResult {
  private final List<String> errors = new ArrayList<>();
  private final List<String> warnings = new ArrayList<>();

  public void addError(String error) {
    errors.add(error);
  }

  public void addWarning(String warning) {
    warnings.add(warning);
  }

  public boolean isValid() {
    return errors.isEmpty();
  }

  @Override
  public String toString() {
    if (isValid()) {
      return "Valid" + (warnings.isEmpty() ? "" : " with warnings: " + warnings);
    }
    return "Invalid: " + errors;
  }
}
